scripts/bootstrap.mk

```diff
@@ -15,104 +16,115 @@ ifeq ($(OS),darwin)
   LIBCRYPTOBOX_ARTIFACT_FILE := libcryptobox.dylib
   LIBCRYPTOBOX_JNI_ARTIFACT_FILE := libcryptobox-jni.dylib
   LIBSODIUM_ARTIFACT_FILE := libsodium.dylib
-  AVS_ARTIFACT_FILE := avs.framework
 endif
 ifeq ($(OS),linux)
   LIBNAME_FLAG=-soname
   LIBCRYPTOBOX_ARTIFACT_FILE := libcryptobox.so
   LIBCRYPTOBOX_JNI_ARTIFACT_FILE := libcryptobox-jni.so
   LIBSODIUM_ARTIFACT_FILE := libsodium.so
-  AVS_ARTIFACT_FILE := avs.framework
 endif
 
-all: install-rust prepare-native cryptobox-c libsodium cryptobox4j avs-download copy-all-libs
+AVS_ARTIFACT_FILE := avs.framework
 
-.PHONY: install-rust
-install-rust:
-	@curl https://sh.rustup.rs -sSf | sh -s -- -y
-	@source "${HOME}/.cargo/env"
+NATIVE_TARBALLS = native/.tarballs
+NATIVE_LIBS = native/libs
+NATIVE := $(NATIVE_LIBS) $(NATIVE_TARBALLS)
+
+CRYPTOBOX_C_URL := https://github.com/wireapp/cryptobox-c/archive/refs/tags/$(CRYPTOBOX_C_VERSION).tar.gz
+CRYPTOBOX_C_TAR_GZ := $(NATIVE_TARBALLS)/cryptobox-c_$(CRYPTOBOX_C_VERSION).tar.gz
+CRYPTOBOX_C_CODE := native/cryptobox-c_$(CRYPTOBOX_C_VERSION)
+CRYPTOBOX_C_ARTIFACT := $(NATIVE_LIBS)/$(LIBCRYPTOBOX_ARTIFACT_FILE)
+
+LIBSODIUM_URL := https://github.com/jedisct1/libsodium/archive/refs/tags/$(LIBSODIUM_VERSION).tar.gz
+LIBSODIUM_TAR_GZ := $(NATIVE_TARBALLS)/libsodium_$(LIBSODIUM_VERSION).tar.gz
+LIBSODIUM_CODE := native/libsodium_$(LIBSODIUM_VERSION)
+LIBSODIUM_ARTIFACT := $(NATIVE_LIBS)/$(LIBSODIUM_ARTIFACT_FILE)
+
+CRYPTOBOX4J_URL = https://github.com/wireapp/cryptobox4j/archive/refs/tags/$(CRYPTOBOX4J_VERSION).tar.gz
+CRYPTOBOX4J_TAR_GZ :=  $(NATIVE_TARBALLS)/cryptobox4j_$(CRYPTOBOX4J_VERSION).tar.gz
+CRYPTOBOX4J_CODE := native/cryptobox4j_$(CRYPTOBOX4J_VERSION)
+CRYPTOBOX4J_ARTIFACT= $(NATIVE_LIBS)/$(LIBCRYPTOBOX_JNI_ARTIFACT_FILE)
+
+AVS_FRAMEWORK_URL := https://github.com/wireapp/wire-avs/releases/download/$(AVS_VERSION)/avs.framework.osx.$(AVS_VERSION).zip
+AVS_FRAMEWORK_ZIP := $(NATIVE_TARBALLS)/avs.framework.osx.$(AVS_VERSION).zip
+AVS_FRAMEWORK_UNZIP := native/avs.framework_$(AVS_VERSION)
+AVS_FRAMEWORK_SRC := $(AVS_FRAMEWORK_UNZIP)/Carthage/Build/iOS/avs.framework
```

Comment:
this is not source code. This is is a binary, probably the name should reflect that.

Reply:
It was here more meant in the sense of source=origin (`cp $src $dst`), still I see the potential for confusion so I changed it to `LOCATION` to reflect it a bit more